### ISSUE TITLE
Make match_scopes more flexible

### ISSUE THREAD

**Issue by rswarbrick-argon** [ArgonDesign/acov#9]
It would probably be better if it looked like this:

    match_scopes "foo" {
      record bar;
      match_scopes "qux" {
         record baz;
      }
    }

**Comment by gezalore** [comment]
Could also just be 
```
in "foo" {
  record bar;
  when baz {
     record qux;
  }
}
```

**Comment by rswarbrick-argon** [comment]
Fixed in 646eeff
